#!/bin/bash

PROJECT=$1
SAMPLE_SHEET=$2
GVCF_LIST=$3
PREFIX=$4

#############FIXED DIRECTORIES################

SCRIPT_DIR="FIND/SCRIPT/HOME/WHEN/READY"
JAVA_1_7="/isilon/sequencing/Kurt/Programs/Java/jdk1.7.0_25/bin"
CORE_PATH="/isilon/sequencing/Seq_Proj"
GATK_DIR="/isilon/sequencing/CIDRSeqSuiteSoftware/gatk/GATK_3/GenomeAnalysisTK-3.3-0"
GATK_3_1_1_DIR="/isilon/sequencing/CIDRSeqSuiteSoftware/gatk/GATK_3/GenomeAnalysisTK-3.1-1"
GATK_DIR_NIGHTLY="/isilon/sequencing/CIDRSeqSuiteSoftware/gatk/GATK_3/GenomeAnalysisTK-nightly-2015-01-15-g92376d3"
SAMTOOLS_DIR="/isilon/sequencing/Kurt/Programs/samtools/samtools-0.1.18/"
TABIX_DIR="/isilon/sequencing/Kurt/Programs/TABIX/tabix-0.2.6/"
CIDR_SEQSUITE_JAVA_DIR="/isilon/sequencing/CIDRSeqSuiteSoftware/java/jre1.7.0_45/bin"
CIDR_SEQSUITE_6_1_1_DIR="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/6.1.1"

##############FIXED FILE PATHS################

KEY="/isilon/sequencing/CIDRSeqSuiteSoftware/gatk/GATK_2/lee.watkins_jhmi.edu.key"
HAPMAP_VCF=/isilon/sequencing/GATK_resource_bundle/2.5/b37/hapmap_3.3.b37.vcf
OMNI_VCF=/isilon/sequencing/GATK_resource_bundle/2.5/b37/1000G_omni2.5.b37.vcf
ONEKG_SNPS_VCF=/isilon/sequencing/GATK_resource_bundle/2.5/b37/1000G_phase1.snps.high_confidence.b37.vcf
DBSNP_138_VCF=/isilon/sequencing/GATK_resource_bundle/2.8/b37/dbsnp_138.b37.vcf
ONEKG_INDELS_VCF=/isilon/sequencing/GATK_resource_bundle/2.2/b37/Mills_and_1000G_gold_standard.indels.b37.vcf
P3_1KG="/isilon/sequencing/1000genomes/Full_Project/Sep_2014/20130502/ALL.wgs.phase3_shapeit2_mvncall_integrated_v5.20130502.sites.vcf.gz"
ExAC="/isilon/sequencing/ExAC/Release_0.3/ExAC.r0.3.sites.vep.vcf.gz"
KNOWN_SNPS=/isilon/sequencing/GATK_resource_bundle/2.8/b37/dbsnp_138.b37.excluding_sites_after_129.vcf
VERACODE_CSV=/isilon/sequencing/CIDRSeqSuiteSoftware/resources/Veracode_hg18_hg19.csv

############################################################################
#################Start of Combine Gvcf Functions############################
############################################################################

## Need to think of another way to do this and handle multiple $PROJECT names for mendel
CREATE_PROJECT_INFO_ARRAY (){
PROJECT_INFO_ARRAY=(`sed 's/\r//g' $SAMPLE_SHEET | sed 's/,/\t/g' | awk 'NR>1 {print $12,$18}' | sort | uniq`)

REF_GENOME=${PROJECT_INFO_ARRAY[0]}
PROJECT_DBSNP=${PROJECT_INFO_ARRAY[1]}
}


COMBINE_GVCF(){
echo \
 qsub \
 -N 'A01_COMBINE_GVCF_'$PROJECT'_chr'$CHROMOSOME \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.A01_COMBINE_GVCF_chr$CHROMOSOME.log \
 $SCRIPT_DIR/A01_COMBINE_GVCF.sh \
 $JAVA_1_7 $GATK_DIR $REF_GENOME \
 $KEY $CORE_PATH $PROJECT $GVCF_LIST \
 $PREFIX $CHROMOSOME
 }

GENOTYPE_GVCF(){
echo \
 qsub \
 -N B02_GENOTYPE_GVCF_$PROJECT"_chr"$CHROMOSOME \
 -hold_jid A01_COMBINE_GVCF_$PROJECT"_chr"$CHROMOSOME \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.B02_GENOTYPE_GVCF_chr$CHROMOSOME.log \
 $SCRIPT_DIR/B02_GENOTYPE_GVCF.sh \
 $JAVA_1_7 $GATK_DIR $REF_GENOME \
 $KEY $CORE_PATH $PROJECT \
 $PREFIX $CHROMOSOME
}

VARIANT_ANNOTATOR(){
echo \
 qsub \
 -N C03_VARIANT_ANNOTATOR_$PROJECT"_chr"$CHROMOSOME \
 -hold_jid B02_GENOTYPE_GVCF_$PROJECT"_chr"$CHROMOSOME \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.C03_VARIANT_ANNOTATOR_chr$CHROMOSOME.log \
 $SCRIPT_DIR/C03_VARIANT_ANNOTATOR.sh \
 $JAVA_1_7 $GATK_DIR $REF_GENOME \
 $KEY $CORE_PATH $PROJECT \
 $PREFIX $CHROMOSOME
}

##############################################################################
#####################End of Combine Gvcf Functions############################
##############################################################################

##############################################################################
##################Start of VQSR and Refinement Functions######################
##############################################################################

GENERATE_COMBINE_VARIANTS_HOLD_ID(){
COMBINE_VARIANTS_HOLD_ID=$COMBINE_VARIANTS_HOLD_ID'C03_VARIANT_ANNOTATOR_'$PROJECT'_chr'$CHROMOSOME','
}

COMBINE_VARIANTS(){
echo \
 qsub \
 -N D04_COMBINE_VARIANTS_$PROJECT \
 -hold_jid $COMBINE_VARIANTS_HOLD_ID \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.D04_COMBINE_VARIANTS.log \
 $SCRIPT_DIR/D04_COMBINE_VARIANTS.sh \
 $JAVA_1_7 $GATK_3_1_1_DIR $REF_GENOME $KEY \
 $CORE_PATH $PROJECT $PREFIX
}

VARIANT_RECALIBRATOR_SNV() {
echo \
 qsub \
 -N E05A_VARIANT_RECALIBRATOR_SNV_$PROJECT \
 -hold_jid D04_COMBINE_VARIANTS_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.E05A_VARIANT_RECALIBRATOR_SNV.log \
 $SCRIPT_DIR/E05A_VARIANT_RECALIBRATOR_SNV.sh \
 $JAVA_1_7 $GATK_DIR $KEY $REF_GENOME $HAPMAP_VCF $OMNI_VCF $ONEKG_SNPS_VCF $DBSNP_138_VCF \
 $CORE_PATH $PROJECT $PREFIX
}


VARIANT_RECALIBRATOR_INDEL() {
echo \
 qsub \
 -N E05B_VARIANT_RECALIBRATOR_INDEL_$PROJECT \
 -hold_jid D04_COMBINE_VARIANTS_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.E05B_VARIANT_RECALIBRATOR_INDEL.log \
 $SCRIPT_DIR/E05B_VARIANT_RECALIBRATOR_INDEL.sh \
 $JAVA_1_7 $GATK_DIR $KEY $REF_GENOME $ONEKG_INDELS_VCF \
 $CORE_PATH $PROJECT $PREFIX
}


APPLY_RECALIBRATION_SNV(){
echo \
 qsub \
 -N F06_APPLY_RECALIBRATION_SNV_$PROJECT \
 -hold_jid E05A_VARIANT_RECALIBRATOR_SNV_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.F06_APPLY_RECALIBRATION_SNV.log \
 $SCRIPT_DIR/F06_APPLY_RECALIBRATION_SNV.sh \
 $JAVA_1_7 $GATK_DIR $KEY $REF_GENOME \
 $CORE_PATH $PROJECT $PREFIX
}


APPLY_RECALIBRATION_INDEL(){
echo \
 qsub \
 -N G07_APPLY_RECALIBRATION_INDEL_$PROJECT \
 -hold_jid F06_APPLY_RECALIBRATION_SNV_$PROJECT','E05B_VARIANT_RECALIBRATOR_INDEL_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.G07_APPLY_RECALIBRATION_INDEL.log \
 $SCRIPT_DIR/G07_APPLY_RECALIBRATION_INDEL.sh \
 $JAVA_1_7 $GATK_DIR $KEY $REF_GENOME \
 $CORE_PATH $PROJECT $PREFIX
}

SELECT_RARE_BIALLELIC(){
echo \
 qsub \
 -N H08A_SELECT_RARE_BIALLELIC_$PROJECT \
 -hold_jid G07_APPLY_RECALIBRATION_INDEL_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.H08A_SELECT_RARE_BIALLELIC.log \
 $SCRIPT_DIR/H08A_SELECT_RARE_BIALLELIC.sh \
 $JAVA_1_7 $GATK_DIR $KEY $REF_GENOME \
 $CORE_PATH $PROJECT $PREFIX
}

ANNOTATE_SELECT_RARE_BIALLELIC(){
echo \
 qsub \
 -N H08A-1_ANNOTATE_SELECT_RARE_BIALLELIC_$PROJECT \
 -hold_jid H08A_SELECT_RARE_BIALLELIC_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.H08A-1_ANNOTATE_SELECT_RARE_BIALLELIC.log \
 $SCRIPT_DIR/H08A-1_ANNOTATE_SELECT_RARE_BIALLELIC.sh \
 $JAVA_1_7 $GATK_DIR $KEY $REF_GENOME \
 $CORE_PATH $PROJECT $PREFIX
}

SELECT_COMMON_BIALLELIC(){
echo \
 qsub \
 -N H08B_SELECT_COMMON_BIALLELIC_$PROJECT \
 -hold_jid G07_APPLY_RECALIBRATION_INDEL_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.H08B_SELECT_COMMON_BIALLELIC.log \
 $SCRIPT_DIR/H08B_SELECT_COMMON_BIALLELIC.sh \
 $JAVA_1_7 $GATK_DIR $KEY $REF_GENOME \
 $CORE_PATH $PROJECT $PREFIX
}

SELECT_MULTIALLELIC(){
echo \
 qsub \
 -N H08C_SELECT_MULTIALLELIC_$PROJECT \
 -hold_jid G07_APPLY_RECALIBRATION_INDEL_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.H08C_SELECT_MULTIALLELIC.log \
 $SCRIPT_DIR/H08C_SELECT_MULTIALLELIC.sh \
 $JAVA_1_7 $GATK_DIR $KEY $REF_GENOME \
 $CORE_PATH $PROJECT $PREFIX
}

COMBINE_VARIANTS_VCF(){
echo \
 qsub \
 -N I09_COMBINE_VARIANTS_VCF_$PROJECT \
 -hold_jid H08C_SELECT_MULTIALLELIC_$PROJECT','H08B_SELECT_COMMON_BIALLELIC_$PROJECT','H08A-1_ANNOTATE_SELECT_RARE_BIALLELIC_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.I09_COMBINE_VARIANTS_VCF.log \
 $SCRIPT_DIR/I09_COMBINE_VARIANTS_VCF.sh \
 $JAVA_1_7 $GATK_3_1_1_DIR $KEY $REF_GENOME \
 $CORE_PATH $PROJECT $PREFIX
}


BGZIP_AND_TABIX_COMBINED_VCF(){
echo \
 qsub \
 -N I09-1_BGZIP_AND_TABIX_COMBINED_VCF_$PROJECT \
 -hold_jid I09_COMBINE_VARIANTS_VCF_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.I09-1_BGZIP_AND_TABIX_COMBINED_VCF.log \
 $SCRIPT_DIR/I09-1_BGZIP_AND_TABIX_COMBINED_VCF.sh \
 $TABIX_DIR \
 $CORE_PATH $PROJECT $PREFIX
}

# ###########################################################################
# #################End of VQSR and Refinement Functions######################
# ###########################################################################

# ###########################################################################
# ###################Start of Vcf Splitter Functions#########################
# ###########################################################################
#
CREATE_SAMPLE_INFO_ARRAY ()
{
SAMPLE_INFO_ARRAY=(`sed 's/\r//g' $SAMPLE_SHEET | awk 'BEGIN{FS=","} NR>1 {print $1,$8,$17,$15,$18,$12}' | sed 's/,/\t/g' | sort -k 2,2 | uniq | awk '$2=="'$SAMPLE'" {print $1,$2,$3,$4,$5,$6}'`)

PROJECT_SAMPLE=${SAMPLE_INFO_ARRAY[0]}
SM_TAG=${SAMPLE_INFO_ARRAY[1]}
TARGET_BED=${SAMPLE_INFO_ARRAY[2]}
TITV_BED=${SAMPLE_INFO_ARRAY[3]}
DBSNP=${SAMPLE_INFO_ARRAY[4]}
SAMPLE_REF_GENOME=${SAMPLE_INFO_ARRAY[5]}
}

SELECT_PASSING_VARIANTS_PER_SAMPLE(){
echo \
 qsub \
 -N J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -hold_jid I09_COMBINE_VARIANTS_VCF_$PROJECT \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.J10_SELECT_VARIANTS_FOR_SAMPLE.log \
 $SCRIPT_DIR/K11_SELECT_VARIANTS_FOR_SAMPLE.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME \
 $CORE_PATH $PROJECT $SM_TAG $PREFIX $PROJECT_SAMPLE
}

BGZIP_AND_TABIX_SAMPLE_VCF(){
echo \
 qsub \
 -N J10-1_BGZIP_AND_TABIX_SAMPLE_VCF_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.J10-1_BGZIP_AND_TABIX_SAMPLE_VCF.log \
 $SCRIPT_DIR/J10-1_BGZIP_AND_TABIX_SAMPLE_VCF.sh \
 $TABIX_DIR \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}

SELECT_VARIANTS_ON_TARGET_BY_SAMPLE(){
echo \
 qsub \
 -N K11A_SELECT_VARIANTS_ON_TARGET_BY_SAMPLE_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11A_SELECT_VARIANTS_ON_TARGET_BY_SAMPLE_.log \
 $SCRIPT_DIR/K11A_SELECT_VARIANTS_ON_TARGET_BY_SAMPLE.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG $TARGET_BED
}

SELECT_SNVS_ON_BAIT_BY_SAMPLE(){
echo \
 qsub \
 -N K11B_SELECT_SNVS_ON_BAIT_BY_SAMPLE_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11B_SELECT_SNVS_ON_BAIT_BY_SAMPLE_.log \
 $SCRIPT_DIR/K11B_SELECT_SNVS_ON_BAIT_BY_SAMPLE.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}

SELECT_SNVS_ON_TARGET_BY_SAMPLE(){
echo \
 qsub \
 -N K11C_SELECT_SNVS_ON_TARGET_BY_SAMPLE_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11C_SELECT_SNVS_ON_TARGET_BY_SAMPLE_.log \
 $SCRIPT_DIR/K11C_SELECT_SNVS_ON_TARGET_BY_SAMPLE.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG $TARGET_BED
}

SELECT_INDELS_ON_BAIT_BY_SAMPLE(){
echo \
 qsub \
 -N K11D_SELECT_INDELS_ON_BAIT_BY_SAMPLE_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11D_SELECT_INDELS_ON_BAIT_BY_SAMPLE_.log \
 $SCRIPT_DIR/K11D_SELECT_INDELS_ON_BAIT_BY_SAMPLE.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}

SELECT_INDELS_ON_TARGET_BY_SAMPLE(){
echo \
 qsub \
 -N K11E_SELECT_INDELS_ON_TARGET_BY_SAMPLE_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11E_SELECT_INDELS_ON_TARGET_BY_SAMPLE_.log \
 $SCRIPT_DIR/K11E_SELECT_INDELS_ON_TARGET_BY_SAMPLE.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG $TARGET_BED
}

SELECT_SNVS_TITV_ALL(){
echo \
 qsub \
 -N K11F_SELECT_SNVS_TITV_ALL_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11F_SELECT_SNVS_TITV_ALL_.log \
 $SCRIPT_DIR/K11F_SELECT_SNVS_TITV_ALL.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG $TITV_BED
}

TITV_ALL(){
echo \
 qsub \
 -N K11F-1_TITV_ALL_$SM_TAG \
 -hold_jid K11F_SELECT_SNVS_TITV_ALL_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11F-1_TITV_ALL.log \
 $SCRIPT_DIR/K11F-1_TITV_ALL.sh \
 $SAMTOOLS_DIR \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}

SELECT_SNVS_TITV_KNOWN(){
echo \
 qsub \
 -N K11G_SELECT_SNVS_TITV_KNOWN_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11G_SELECT_SNVS_TITV_KNOWN.log \
 $SCRIPT_DIR/K11G_SELECT_SNVS_TITV_KNOWN.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME $KNOWN_SNPS \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG $TITV_BED
}

TITV_KNOWN(){
echo \
 qsub \
 -N K11G-1_TITV_KNOWN_$SM_TAG \
 -hold_jid K11G_SELECT_SNVS_TITV_KNOWN_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11G-1_TITV_KNOWN.log \
 $SCRIPT_DIR/K11G-1_TITV_KNOWN.sh \
 $SAMTOOLS_DIR \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}

SELECT_SNVS_TITV_NOVEL(){
echo \
 qsub \
 -N K11H_SELECT_SNVS_TITV_NOVEL_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11H_SELECT_SNVS_TITV_NOVEL.log \
 $SCRIPT_DIR/K11H_SELECT_SNVS_TITV_NOVEL.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME $KNOWN_SNPS \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG $TITV_BED
}

TITV_NOVEL(){
echo \
 qsub \
 -N K11H-1_TITV_NOVEL_$SM_TAG \
 -hold_jid K11H_SELECT_SNVS_TITV_NOVEL_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11H-1_TITV_NOVEL.log \
 $SCRIPT_DIR/K11H-1_TITV_NOVEL.sh \
 $SAMTOOLS_DIR \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}

CONCORDANCE_ON_TARGET_PER_SAMPLE(){
echo \
 qsub \
 -N L12_CONCORDANCE_ON_TARGET_PER_SAMPLE_$SM_TAG \
 -hold_jid K11F-1_TITV_ALL_$SM_TAG,K11G-1_TITV_KNOWN_$SM_TAG,K11H-1_TITV_NOVEL_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.L12_CONCORDANCE_ON_TARGET_PER_SAMPLE.log \
 $SCRIPT_DIR/L12_CONCORDANCE_ON_TARGET_PER_SAMPLE.sh \
 $CIDR_SEQSUITE_JAVA_DIR $CIDR_SEQSUITE_6_1_1_DIR $VERACODE_CSV \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG $TARGET_BED
}

HC_BED_GENERATION(){
echo \
 qsub \
 -N K11I_HC_BED_GENERATION_$SM_TAG \
 -hold_jid J10_SELECT_VARIANTS_FOR_SAMPLE_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11I_HC_BED_GENERATION.log \
 $SCRIPT_DIR/K11I_HC_BED_GENERATION.sh \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}


HC_BAM_GENERATION(){
echo \
 qsub \
 -N K11I-1_HC_BAM_GENERATION_$SM_TAG \
 -hold_jid K11I_HC_BED_GENERATION_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11I-1_HC_BAM_GENERATION.log \
 $SCRIPT_DIR/K11I-1_HC_BAM_GENERATION.sh \
 $JAVA_1_7 $GATK_DIR $KEY $SAMPLE_REF_GENOME $DBSNP \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}

BGZIP_AND_TABIX_HC_VCF(){
echo \
 qsub \
 -N K11I-1-1_BGZIP_AND_TABIX_HC_VCF_$SM_TAG \
 -hold_jid K11I-1_HC_BAM_GENERATION_$SM_TAG \
 -o $CORE_PATH/$PROJECT/LOGS/$PROJECT.$SM_TAG.K11I-1-1_BGZIP_AND_TABIX_HC_VCF.log \
 $SCRIPT_DIR/K11I-1-1_BGZIP_AND_TABIX_HC_VCF.sh \
 $TABIX_DIR \
 $CORE_PATH $PROJECT_SAMPLE $SM_TAG
}

##########################################################################
######################End of Functions####################################
##########################################################################

CREATE_PROJECT_INFO_ARRAY

for CHROMOSOME in {{1..22},{X,Y}};
 do
COMBINE_GVCF
GENOTYPE_GVCF
VARIANT_ANNOTATOR
GENERATE_COMBINE_VARIANTS_HOLD_ID
 done

COMBINE_VARIANTS
VARIANT_RECALIBRATOR_SNV
VARIANT_RECALIBRATOR_INDEL
APPLY_RECALIBRATION_SNV
APPLY_RECALIBRATION_INDEL
SELECT_RARE_BIALLELIC
ANNOTATE_SELECT_RARE_BIALLELIC
SELECT_COMMON_BIALLELIC
SELECT_MULTIALLELIC
COMBINE_VARIANTS_VCF
BGZIP_AND_TABIX_COMBINED_VCF


for SAMPLE in $(awk 'BEGIN {FS=","} NR>1 {print $8}' $SAMPLE_SHEET | sort | uniq )
do
CREATE_SAMPLE_INFO_ARRAY
SELECT_PASSING_VARIANTS_PER_SAMPLE
BGZIP_AND_TABIX_SAMPLE_VCF
SELECT_VARIANTS_ON_TARGET_BY_SAMPLE
SELECT_SNVS_ON_BAIT_BY_SAMPLE
SELECT_SNVS_ON_TARGET_BY_SAMPLE
SELECT_INDELS_ON_BAIT_BY_SAMPLE
SELECT_INDELS_ON_TARGET_BY_SAMPLE
SELECT_SNVS_TITV_ALL
TITV_ALL
SELECT_SNVS_TITV_KNOWN
TITV_KNOWN
SELECT_SNVS_TITV_NOVEL
TITV_NOVEL
HC_BED_GENERATION
HC_BAM_GENERATION
BGZIP_AND_TABIX_HC_VCF
CONCORDANCE_ON_TARGET_PER_SAMPLE
done



# Chromosome loop for the COMBINE_GVCF (REPLACES SUBMITTER_COMBINE_GVCF.sh and COMBINE_GVCF.sh)
# 	Breakout COMBINE GVCF to:
# 			   COMBINE_GVCF
# 			   GENOTYPE_GVCF    (DEPENDENT ON COMBINE_GVCF)
# 			   VARIANT_ANNOTATOR (DEPENDENT ON GENOTYPE_GVCF)
#
# ### Next breakdown VQSR.GATK-3.3-0_CGP_26April2015.sh
# 		CATVARIANTS
# 		VARIANT_RECAL_SNV (DEPENDENT ON CATVARIANTS)
# 		VARIANT_RECAL_INDEL (DEPENDENT ON CATVARIANTS)
# 		APPLY_SNV (DEPENDENT ON VARIANT_RECAL_SNV)
# 		APPLY_INDEL (DEPENDENT ON VARIANT_RECA_INDEL && APPLY_SNV)
# 		BGZIP&TABIX (DEPENDENT ON APPLY_INDEL)
# 		CALCULATE_GENOTYPE_POSTERIORS (DEPENDENT ON APPLY_INDEL)
# 		VARIANT_ANNOTATOR (CALCULATE_GENOTYPE_POSTERIORS)
# 		BGZIP&TABIX (DEPENDENT ON VARIANT_ANNOTATOR)
#
# ### Next breakdown of VCF.Splitter.NEW.PIPELINE.HC.bam.sh (Also to replace the SUBMITTER)
# 		## Note: All these files can either:
# 							# A: Run off of the refined VQSR (big vcf)
# 							# B: Run off the 1st step vcf that has the extracted the passing variants for one sample ( I like this more to try 1st)
# 		SELECT_PASSING_VARIANTS (DEPENDENT ON CALCULATE_GENOTYPE_POSTERIORS)
# 		BGZIP_AND_TABIX (DEPENDENT ON SELECT_PASSING_VARIANTS)
# 		PASSING_VARIANTS_ON_TARGET_BY_SAMPLE (DEPENDENT ON COMBINE_VARIANTS OR CALCULATE_GENOTYPE_POSTERIORS)
# 		PASSING_SNVS_ON_BAIT_BY_SAMPLE (DEPENDENT ON COMBINE_VARIANTS OR CALCULATE_GENOTYPE_POSTERIORS)
# 		PASSING_SNVS_ON_TARGET_BY_SAMPLE (DEPENDENT ON COMBINE_VARIANTS OR CALCULATE_GENOTYPE_POSTERIORS)
# 		PASSING_INDELS_ON_BAIT_BY_SAMPLE (DEPENDENT ON COMBINE_VARIANTS OR CALCULATE_GENOTYPE_POSTERIORS)
# 		PASSING_INDELS_ON_TARGET_BY_SAMPLE (DEPENDENT ON COMBINE_VARIANTS OR CALCULATE_GENOTYPE_POSTERIORS)
# 		PASSING_SNVS_TITV_ALL (DEPENDENT ON COMBINE_VARIANTS OR CALCULATE_GENOTYPE_POSTERIORS)
# 		TITV_ALL (DEPENDENT ON PASSING_SNVS_TITV_ALL)
# 		PASSING_SNVS_TITV_KNOWN (DEPENDENT ON COMBINE_VARIANTS OR CALCULATE_GENOTYPE_POSTERIORS)
# 		TITV_KNOWN (DEPENDENT ON PASSING_SNVS_TITV_KNOWN)
# 		PASSING_SNVS_TITV_NOVEL (DEPENDENT ON COMBINE_VARIANTS OR CALCULATE_GENOTYPE_POSTERIORS)
# 		TITV_NOVEL (DEPENDENT ON PASSING_SNVS_TITV_NOVEL)
# 		CONCORDANCE_ON_TARGET_PER_SAMPLE (DEPENDENT ON TITV_NOVEL,TITV_KNOWN,TITV_ALL)